src/mdesktopaction.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+** This file was derived from the MDesktopAction implementation in the
+** libmeegotouch library.
+**
+** Copyright (C) 2021 Chupligin Sergey (NeoChapay) <neochapay@gmail.com>
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file LICENSE.LGPL included in the packaging
+** of this file.
+**
+****************************************************************************/
+
+#ifndef MDESKTOPACTION_H
+#define MDESKTOPACTION_H
+
+#include <mlite-global.h>
+#include <QString>
+
+class MDesktopEntryPrivate;
+class MDesktopActionPrivate;
+
+class MLITESHARED_EXPORT MDesktopAction
+{
+
+public:
+    MDesktopAction(const QString &fileName, const QString &action);
```

Comment:
Is there need for such ctor? That is expecting app developers to use it? Would myself assume the main thing is the getter desktop entry. If so, this could be even private + desktop entry as friend.

For another thing, the fact that all the actions read the file separately is still silly.